File: csharp/DVSClient/DVSClient/address/format/LineContent.cs
using DVSClient.Common;

namespace DVSClient.Address.Format
{
    public enum LineContent
    {
        [EnumStringValue("none")]
        None,
        [EnumStringValue("address")]
        Address,
        [EnumStringValue("name")]
        Name,
        [EnumStringValue("ancillery")]
        Ancillery,
        [EnumStringValue("dataplus")]
        Dataplus
    }
}

File: java/src/main/java/com/experian/dvs/client/address/format/LineContent.java
package com.experian.dvs.client.address.format;

import java.util.HashMap;
import java.util.Map;

public enum LineContent {

    NONE("none"),
    ADDRESS("address"),
    NAME("name"),
    ANCILLERY("ancillery"),
    DATAPLUS("dataplus")
    ;

    private final String name;
    //Map of name to AddressConfidence
    private static final Map<String, LineContent> nameToLineContent = new HashMap<>();
    static {
        for (LineContent lineContent : LineContent.values()) {
            nameToLineContent.put(lineContent.name, lineContent);
        }
    }

    LineContent(final String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static LineContent fromName(final String name) {
        return nameToLineContent.get(name);
    }
}
